R/csv.R

```diff
@@ -297,41 +270,74 @@ read_cmdstan_csv <- function(files,
     repaired_variables <- gsub("log_p__", "lp__", repaired_variables)
     repaired_variables <- gsub("log_g__", "lp_approx__", repaired_variables)
   }
-
   model_param_dims <- variable_dims(metadata$model_params)
   metadata$stan_variable_dims <- model_param_dims
   metadata$stan_variables <- names(model_param_dims)
 
   if (metadata$method == "sample") {
-    warmup_draws <- bind_list_of_draws_array(warmup_draws)
-    if (!is.null(warmup_draws)) {
+    if (length(warmup_draws) > 0) {
+      warmup_draws <- posterior::as_draws_array(warmup_draws)
       posterior::variables(warmup_draws) <- repaired_variables
+      if (posterior::niterations(warmup_draws) == 0) {
+        warmup_draws <- NULL
+      }
+    } else {
+      warmup_draws <- NULL
+    }
+    if (length(draws) > 0) {
+      draws <-  posterior::as_draws_array(draws)
+      posterior::variables(draws) <- repaired_variables
```

Comment:
This line will error if draws are read in for particular variables instead of all variables:

```r
fit <- cmdstanr_example()
fit$draws("alpha")
```

gives this error: 

```
Error in dimnames(x) <- `*vtmp*` : 
  length of 'dimnames' [3] not equal to array extent
Called from: `variables<-.draws_array`(`*tmp*`, value = "alpha")
```

When the error happens `posterior::variables(draws)` has both `"lp__"` and `"alpha"` whereas `repaired_variables` is just `"alpha"`. I guess this is because `"lp__"` has already been read in immediately after fitting? 

This same error occurs a few lines earlier at the analogous line for warmup draws if warmup is included.

Reply:
Thanks, the issue was the missing , drop = FALSE, which caused names to drop. Should be good now and tests seem to be passing.